src/library_fs.js

```diff
@@ -1132,9 +1132,6 @@ FS.staticInit();` +
       return stream.position;
     },
     read: (stream, buffer, offset, length, position) => {
-#if CAN_ADDRESS_2GB
-      offset >>>= 0;
-#endif
```

Comment:
What do you mean "can arrive from user input"?    This function gets call from native code via the syscall layer, and this change ensure we treat them as unsigned?

Or are you talking about a JS user calling this code with an out-of-range offset?  e.g. -1?    If a user did call this function with a negative number then I don't this we want to cooerce into a positive one.. we want it to result in an actually.. the call should fail with invalid offset later on this function function.    

Using `>>>= 0` on a random JS value passed by the user seems wrong and potentially dangerous since it looses information and could do the wrong thing.

Reply:
I mean, a user can write `FS.read(..)` in their JS code. But yeah, maybe we don't need to fix up user input like that. It's possibly a breaking change, but seems like an acceptable risk.

Maybe an assertion on the offset being `>=0` would help users that do have such input that used to work?

Reply:
Yeah I was thinking the same thing.

Reply:
Actually I'm not sure it even makes sense to check for these kinds of negative offset here.. the offset parameter comes from a uint64 parameter to wasi_fd_pwrite.   So its desn't come through as an i32, but goes through the i53 number conversion from i64.     Large values won't show as a negative in the same way.

Reply:
> the offset parameter comes from a uint64 parameter to wasi_fd_pwrite.

But, again, users can write `FS.read(..)` by hand?

An assertion could help figure out bugs in such code. I don't think there is much cost to adding an assertion in debug builds, and it can help here.

Reply:
I am happy to add the assert, because why not have more assert right.   But I logically don't see how the offset being passed here could end up be negative due to an u32 being interpreted as an i32.

Remember this is not a pointer we are talking about here, but a file offset, which is i64.. and never i32.

If they called a native function to get the offset it would be returning i64 not i32.    Such users would have to be dealing with files that have offsets above 2gb... so approximating offset as an i32 I likely not something they would be doing.

Also remember that this function already has to handle invalid offset so they would already get a error returns.

Reply:
Added asserts